ams/AssetAnalyzer.cs

```diff
@@ -132,63 +134,109 @@ await foreach (var locator in locators!)
         public override async Task MigrateAsync(CancellationToken cancellationToken)
         {
             var watch = Stopwatch.StartNew();
-            _logger.LogInformation("Begin analysis of assets for account: {name}", _analysisOptions.AccountName);
-            var account = await GetMediaAccountAsync(_analysisOptions.AccountName, cancellationToken);
-            double totalAssets = await QueryMetricAsync(account.Id.ToString(), "AssetCount", cancellationToken);
-            _logger.LogInformation("The total asset count of the media account is {count}.", totalAssets);
+            _logger.LogInformation("Begin analysis of items for account: {name}", _analysisOptions.AccountName);
+
+            var reportGenerator = new ReportGenerator(_globalOptions.HtmlReportFile, _globalOptions.JsonReportFile, _logger);
+            reportGenerator.WriteHeader();
 
             var resourceFilter = GetAssetResourceFilter(_analysisOptions.ResourceFilter,
                                                         _analysisOptions.CreationTimeStart,
                                                         _analysisOptions.CreationTimeEnd);
 
-            var reportGenerator = new ReportGenerator(_globalOptions.HtmlReportFile, _globalOptions.JsonReportFile, _logger);
-            reportGenerator.WriteHeader();
-
-            await _resourceProvider.SetStorageResourceGroupsAsync(account, cancellationToken);
-            var assets = account.GetMediaAssets()
-                .GetAllAsync(resourceFilter, cancellationToken: cancellationToken);
             var statistics = new AssetStats();
             var assetTypes = new ConcurrentDictionary<string, int>();
 
-            List<MediaAssetResource>? filteredList = null;
-
-            if (resourceFilter != null)
+            if (_analysisOptions.IsStorageAcc)
             {
-                // When a filter is used, it usually include a small list of assets,
-                // The total count of asset can be extracted in advance without much perf hit.
-                filteredList = await assets.ToListAsync(cancellationToken);
+                var (storageClient, accountId) = await _resourceProvider.GetStorageAccount(_analysisOptions.AccountName, cancellationToken);
 
-                totalAssets = filteredList.Count;
-            }
+                double totalItems = await GetStorageBlobMetricAsync(accountId, cancellationToken);
+                var containers = storageClient.GetBlobContainersAsync(
+                              prefix: _analysisOptions.Prefix, cancellationToken: cancellationToken);
```

Comment:
The prefix can be extracted from the resource-filter option, so that  no need to add a new --prefix option for this change.